Code/Framework/AzToolsFramework/AzToolsFramework/Application/ToolsApplication.cpp

```diff
@@ -385,7 +386,10 @@ namespace AzToolsFramework
         SliceUtilities::Reflect(context);
         Prefab::PrefabIntegrationManager::Reflect(context);
 
-        ComponentModeFramework::ComponentModeDelegate::Reflect(context);
+        if (!ComponentSwitcherEnabled())
+        {
+            ComponentModeFramework::ComponentModeDelegate::Reflect(context);
+        }
```

Comment:
update: This might be okay for now, but we probably also want (maybe as a follow-up PR) to actually go and remove all the EditContext instances of the button for the `ComponentModeDelegate`. I'm a little worried we do still need the SerializeReflect call for this, just not the `EditContext` (I think removing this might break the double click to enter ComponentMode potentially? :thinking:) It would be nicer to remove just the EditContext part if possible. Maybe we could catch-up about this offline just to check nothing might break (removing serialize context stuff is always risky I'm afraid 😝)